atom/browser/net/url_request_stream_job.cc

```diff
@@ -75,11 +82,13 @@ void URLRequestStreamJob::BeforeStartInUI(v8::Isolate* isolate,
     return;
   }
 
-  subscriber_.reset(new mate::EventSubscriber<URLRequestStreamJob>(
-      this, isolate, data.GetHandle()));
-  subscriber_->On("data", &URLRequestStreamJob::OnData);
-  subscriber_->On("end", &URLRequestStreamJob::OnEnd);
-  subscriber_->On("error", &URLRequestStreamJob::OnError);
+  subscriber_.reset(new mate::EventSubscriber(isolate, data.GetHandle()));
+  subscriber_->On("data", base::Bind(&URLRequestStreamJob::OnData,
+                                     weak_factory_.GetWeakPtr()));
```

Comment:
So if I understood correctly, the  callback is thread-safe and will synchronize internally if the URLJob happens to be destroyed at the same time the event is being emitted?